demo/demo.tsx

```diff
@@ -0,0 +1,254 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+import {
+    Router,
+    Routes,
+    Route,
+    useIsRouting,
+    useSearchParams,
+} from '@solidjs/router';
+import { transform } from 'enketo-transformer/web';
+import { createEffect, createResource, createSignal, on, Show } from 'solid-js';
+import { For, render } from 'solid-js/web';
+import { fixtures as baseFixtures } from '../test/shared';
+
+const fixtures = baseFixtures.sort((A, B) => {
+    const a = A.fileName.toLowerCase().replace(/.*\/([^/]+)$/, '$1');
+    const b = B.fileName.toLowerCase().replace(/.*\/([^/]+)$/, '$1');
+
+    if (a > b) {
+        return 1;
+    }
+
+    return b > a ? -1 : 0;
+});
+
+const xformsByFileName = new Map(
+    fixtures.map((fixture) => [fixture.fileName, fixture.xform])
+);
+
+function Demo() {
+    const [params, setParams] = useSearchParams();
+    const isRouting = useIsRouting();
+    const [fileName, setFileName] = createSignal<string | null>(params.xform);
```

Comment:
It's very likely this is not an ideal way to use Solid's reactivity or Solid Router. This is the first time I've actually put any real Solid code into an actual PR, and I'm hoping to get more familiar with its idioms in the future. But for now it was a quick refactor of a much sloppier pure DOM version of the same demo.